graveler/graveler.go

```diff
@@ -1043,16 +1049,24 @@ func (g *Graveler) ResetPrefix(ctx context.Context, repositoryID RepositoryID, b
 // Example: consider the following tree: C1 -> C2 -> C3, with the branch pointing at C3.
 // To revert C2, we merge C1 into the branch, with C2 as the merge base.
 // That is, try to apply the diff from C2 to C1 on the tip of the branch.
-func (g *Graveler) Revert(ctx context.Context, repositoryID RepositoryID, branchID BranchID, ref Ref, committer string, message string, metadata Metadata) (CommitID, error) {
-	commitID, err := g.branchLocker.MetadataUpdater(ctx, repositoryID, branchID, func() (interface{}, error) {
-		commitRecord, err := g.getCommitRecordFromRef(ctx, repositoryID, ref)
-		if err != nil {
-			return "", fmt.Errorf("get commit from ref %s: %w", ref, err)
-		}
-		if len(commitRecord.Parents) > 1 {
-			// TODO(johnnyaug) support reverting merge commits
-			return "", ErrRevertMergeCommit
+// If the commit is a merge commit, 'parentNumber' is the parent number (1-based) relative to which the revert is done.
+func (g *Graveler) Revert(ctx context.Context, repositoryID RepositoryID, branchID BranchID, ref Ref, parentNumber int, commitParams CommitParams) (CommitID, error) {
+	commitRecord, err := g.getCommitRecordFromRef(ctx, repositoryID, ref)
+	if err != nil {
+		return "", fmt.Errorf("get commit from ref %s: %w", ref, err)
+	}
+	if len(commitRecord.Parents) > 1 && parentNumber <= 0 {
+		// if commit has more than one parent, must explictly specify parent number
+		return "", ErrRevertMergeNoParent
+	}
+	if parentNumber > 0 {
+		// validate parent is in range:
+		if parentNumber > len(commitRecord.Parents) { // parent number is 1-based
```

Comment:
>= or do -- before

Reply:
@nopcoder didn't want to `--` before, because I use the number for the error
`>` is valid here, I think.

Reply:
My bad